---
title: C28105 | Microsoft-Dokumentation
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
caps.latest.revision: 5
author: corob-msft
ms.author: corob
manager: jillfra
ms.openlocfilehash: c985cc9c4820c02ea427987aca1703beec2e97f1
ms.sourcegitcommit: 6cfffa72af599a9d667249caaaa411bb28ea69fd
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 09/02/2020
ms.locfileid: "77266314"
---
# <a name="c28105"></a>C28105
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Warnung C28105: Verlust von Ressource aufgrund einer Ausnahme  
  
 Die angegebene Ressource wird nicht freigegeben, wenn eine Ausnahme ausgelöst wird. Die vom Pfad angegebene Anweisung kann eine Ausnahme auslöst. Diese Warnung ähnelt der Warnung [C28103](../code-quality/c28103.md), mit dem Unterschied, dass in diesem Fall eine Ausnahme beteiligt ist.  
  
## <a name="example"></a>Beispiel  
 Im folgenden Codebeispiel wird diese Warnung generiert:  
  
```  
res = KeSaveFloatingPointState(buffer);  
  
res = AllocateResource(Resource);  
char *p2 = new char[10]; // could throw  
  
delete[] p2;  
FreeResource(Resource)  
```  
  
 Im folgenden Codebeispiel wird diese Warnung vermieden:  
  
```  
res = AllocateResource(Resource);  
char *p2;  
  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
FreeResource(Resource)  
```
